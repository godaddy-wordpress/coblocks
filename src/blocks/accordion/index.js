/**
 * External dependencies
 */
import { AccordionIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	/* translators: block description */
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			count: 2,
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'tabs', 'coblocks' ),
		/* translators: block keyword (abbreviation for "frequently asked questions") */
		__( 'faq', 'coblocks' ),
	],
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	/* translators: block name */
	title: __( 'Accordion', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
