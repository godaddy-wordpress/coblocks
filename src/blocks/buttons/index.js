/**
 * THIS BLOCK IS DEPRECRATED. IT WILL NOT SHOW UP IN THE INSERTER ANYMORE.
 * SEE /src/js/deprecations/deprecate-coblocks-buttons.js
 */

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { button as icon } from '@wordpress/icons';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Buttons', 'coblocks' ),
	/* translators: block description */
	description: __( 'Prompt visitors to take action with multiple buttons, side by side.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'link', 'coblocks' ),
		/* translators: block keyword (abbreviation for "call to action") */
		__( 'cta', 'coblocks' ),
		/* translators: block keyword */
		__( 'call to action', 'coblocks' ),
	],
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
