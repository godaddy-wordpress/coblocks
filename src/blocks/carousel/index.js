/**
 * WordPress dependencies
 */
import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';

/**
 * Internal dependencies
 */
import BlockEdit from './block-edit';
import metadata from './block.json';

const settings = {
	edit: BlockEdit,
	save: () => (
		<div { ...useBlockProps.save() }>
			<InnerBlocks.Content />
		</div>
	),
};

const name = metadata.name;
export { name, metadata, settings };
