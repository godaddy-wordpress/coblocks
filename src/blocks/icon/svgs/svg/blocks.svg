<svg fill="none" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="m10.3519669 11.7141378v8.2858622h8.2815735v-8.2858622zm-10.3519669 8.2858622h8.2815735v-8.2858622h-8.2815735zm0-18.64319006v8.28586225h8.2815735v-8.28586225zm14.1407867-1.35680994-5.8592132 5.85189021 5.8592132 5.86224759 5.8592133-5.86224759z" fill="currentColor" /></svg>