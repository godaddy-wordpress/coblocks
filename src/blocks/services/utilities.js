/**
 * External dependencies.
 */
import { find } from 'lodash';

/**
 * Internal dependencies.
 */
import icons from './icons';
import { hasEmptyAttributes } from '../../utils/block-helpers';
import fromEntries from '../../js/coblocks-fromEntries';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import TokenList from '@wordpress/token-list';

/**
 * Constants
 */
const SERVICES_ALLOWED_BLOCKS = [ 'coblocks/service' ];
const SERVICES_TEMPLATE = [ [ 'coblocks/service' ] ];
const SERVICE_ALLOWED_BLOCKS = [ 'core/heading', 'core/buttons', 'core/paragraph' ];

const layoutOptions = [
	{
		name: 'threebyfour',
		label: '4:3',
		icon: icons.service43,
		isDefault: true,
	},
	{
		name: 'sixbynine',
		label: '16:9',
		icon: icons.service169,
	},
	{
		name: 'square',
		label: __( 'Square', 'coblocks' ),
		icon: icons.serviceSquare,
	},
	{
		name: 'circle',
		label: __( 'Circle', 'coblocks' ),
		icon: icons.serviceCircle,
		defaultAlign: 'center',
	},
];

/**
 * Returns the active style from the given className.
 *
 * @param {Array}  styles    Block style variations.
 * @param {string} className Class name
 * @return {Object?} The active style.
 */
function getActiveStyle( styles, className ) {
	for ( const style of new TokenList( className ).values() ) {
		if ( style.indexOf( 'is-style-' ) === -1 ) {
			continue;
		}

		const potentialStyleName = style.substring( 9 );
		const activeStyle = find( styles, { name: potentialStyleName } );

		if ( activeStyle ) {
			return activeStyle;
		}
	}

	return find( styles, 'isDefault' );
}

/**
 * Replaces the active style in the block's className.
 *
 * @param {string}  className   Class name.
 * @param {Object?} activeStyle The replaced style.
 * @param {Object}  newStyle    The replacing style.
 * @return {string} The updated className.
 */
function replaceActiveStyle( className, activeStyle, newStyle ) {
	const list = new TokenList( className );

	if ( activeStyle ) {
		list.remove( 'is-style-' + activeStyle.name );
	}

	list.add( 'is-style-' + newStyle.name );

	return list.value;
}

/**
 * Given an attributes object will return true if attributes are unset.
 *
 * @param {Object} attributes The replacing style.
 * @return {boolean} Whether or not the attributes object is empty.
 */
const isEmpty = ( attributes ) => {
	const attributesToCheck = [ 'columns', 'buttons' ];
	const newAttributes = Object.entries( attributes ).filter( ( [ key ] ) =>
		attributesToCheck.includes( key )
	);

	return hasEmptyAttributes( fromEntries( newAttributes ) );
};

/**
 * Given an innerBlocks array will return true if innerBlocks attributes are unset.
 *
 * @param {Array} innerBlocks Array of innerBlocks.
 * @return {boolean} Whether or not the innerBlocks attributes object is empty.
 */
const isEmptyInnerBlocks = ( innerBlocks ) => {
	//innerBlock attributes to search for.
	const attributesToCheck = [ 'content' ];

	let blocksAreEmpty = true;
	innerBlocks.forEach( ( block ) => {
		const newAttributes = Object.entries( block.attributes ).filter( ( [ key ] ) =>	attributesToCheck.includes( key ) );

		if ( ! hasEmptyAttributes( fromEntries( newAttributes ) ) ) {
			blocksAreEmpty = false;
		}
	} );

	return blocksAreEmpty;
};

export {
	replaceActiveStyle,
	getActiveStyle,
	isEmpty,
	isEmptyInnerBlocks,
	layoutOptions,
	SERVICES_ALLOWED_BLOCKS,
	SERVICES_TEMPLATE,
	SERVICE_ALLOWED_BLOCKS,
};
