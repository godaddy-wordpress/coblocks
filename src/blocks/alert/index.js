/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const { replaceBlocks } = dispatch( 'core/block-editor' );

		replaceBlocks(
			[ props.clientId ],
			createBlock( 'core/paragraph', props.attributes ),
		);
		return null;
	},
	parent: [],
	save: () => null,
	title: metadata.title,
};

export { name, metadata, settings };
