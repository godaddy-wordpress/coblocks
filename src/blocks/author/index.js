/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch, select } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const parentBlock = select( 'core/editor' ).getBlocksByClientId( props.clientId )[ 0 ];
		if ( ! parentBlock ) {
			return null;
		}

		// Used to set text color props
		const hasTextColor = ( props.attributes.hasOwnProperty( 'textColor' ) );
		const hasCustomTextColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'text' ) );

		// Used to set background color props
		const hasBackgroundColor = ( props.attributes.hasOwnProperty( 'backgroundColor' ) );
		const hasCustomBackgroundColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'background' ) );

		// Used to set font props
		const hasFontSize = ( props.attributes.hasOwnProperty( 'fontSize' ) );
		const hasCustomFont = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'typography' ) );

		const { replaceBlocks } = dispatch( 'core/block-editor' );
		const isRTL = select( 'core/editor' ).getEditorSettings().isRTL;

		const authorNameBlock = () => {
			let authorNameBlockProps = {
				content: props.attributes.name,
				level: 4,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			// Headings do not inherit color from parent - we descend the style here for initial match.
			if ( hasCustomTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						color: { ...authorNameBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			return createBlock( 'core/paragraph', authorNameBlockProps );
		};

		const authorBioBlock = () => {
			let authorBioBlockProps = {
				content: props.attributes.biography,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					style: {
						...authorBioBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			return createBlock( 'core/paragraph', authorBioBlockProps );
		};

		const buttonBlock = parentBlock.innerBlocks;
		// Set buttons to align.
		buttonBlock.forEach( function( button, index ) {
			buttonBlock[ index ].attributes.align = isRTL ? 'right' : 'left';
		} );
		// Button blocks are no-longer stand-alone and should be within `core/buttons` block.
		const coreButtons = createBlock( 'core/buttons', { }, [ ...buttonBlock ] );

		const imageBlock = createBlock( 'core/image', { className: 'is-style-rounded', id: props.attributes.imgId, url: props.attributes.imgUrl } );

		const leftColumn = createBlock( 'core/column', { width: '25%' }, [ imageBlock ] );
		const rightColumn = createBlock( 'core/column', { width: '75%' }, [ authorNameBlock(), authorBioBlock(), coreButtons ] );

		const columnsBlock = () => {
			let columnsBlockProps = {
				...( props.attributes.hasOwnProperty( 'className' ) && { className: props.attributes.className } ),
				...( props.attributes.hasOwnProperty( 'animation' ) && { animation: props.attributes.animation } ),
				...( props.attributes.hasOwnProperty( 'anchor' ) && { anchor: props.attributes.anchor } ),
				style: {
					spacing: {
						padding: {
							bottom: '2.5rem',
							left: '2.5rem',
							right: '2.5rem',
							top: '2.5rem',
						},
					},
				},
			};

			// Migrate text colors into columns block props.
			if ( hasCustomTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			// Migrate background colors into columns block props.
			if ( hasCustomBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: props.attributes.style.color.background,
						},
					},
				};
			}
			if ( hasBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					backgroundColor: props.attributes.backgroundColor,
				};
			}

			// Does not have background color set. Use default block color from CoBlocks.
			const hasNoBackgroundColorSet = ! props.attributes.hasOwnProperty( 'backgroundColor' ) &&
				! props.attributes?.style?.color?.hasOwnProperty( 'background' );
			if ( hasNoBackgroundColorSet ) {
				// $dark-opacity-light-200 computed #8C8C971A - no variable access in root.
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: '#8C8C971A',
						},
					},
				};
			}

			return createBlock( 'core/columns', columnsBlockProps, [ leftColumn, rightColumn ] );
		};

		replaceBlocks(
			[ props.clientId ],
			columnsBlock()
		);

		return null;
	},
	parent: [],
	save: () => null,
	/* translators: block name */
	title: metadata.title,
};

export { name, metadata, settings };
