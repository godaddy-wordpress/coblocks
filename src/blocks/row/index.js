/**
 * External dependencies
 */
import { RowIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon, Spinner } from '@wordpress/components';
import { lazy, Suspense } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { BackgroundAttributes } from '../../components/background';
import deprecated from './deprecated';
import DimensionsAttributes from '../../components/dimensions-control/attributes';
const Edit = lazy( () => import( './edit' ) );
import { getEditWrapperProps } from './utilities';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';
import variations from './variations';

/**
 * Block constants
 */
const { name, category } = metadata;

const attributes = {
	...DimensionsAttributes,
	...BackgroundAttributes,
	...metadata.attributes,
};

const settings = {
	/* translators: block name */
	title: __( 'Row', 'coblocks' ),
	/* translators: block description */
	description: __( 'Add a structured wrapper for column blocks, then add content blocks youâ€™d like to the columns.', 'coblocks' ),
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'rows', 'coblocks' ),
		/* translators: block keyword */
		__( 'columns', 'coblocks' ),
		/* translators: block keyword */
		__( 'layouts', 'coblocks' ),
	],
	supports: {
		align: [ 'wide', 'full' ],
		anchor: true,
		gutter: {
			default: 'medium',
			customDefault: 0.8,
		},
		stackedOnMobile: true,
		coBlocksSpacing: true,
	},
	attributes,
	variations,
	transforms,
	edit: ( props ) => (
		<Suspense fallback={ <Spinner /> }>
			<Edit { ...props } />
		</Suspense>
	),
	example: {
		attributes: {},
	},
	getEditWrapperProps,
	save,
	deprecated,
};

export { name, category, metadata, settings };
