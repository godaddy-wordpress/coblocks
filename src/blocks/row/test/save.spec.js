/* eslint-disable jest/no-conditional-expect */
/**
 * External dependencies
 */
import { createBlock, registerBlockType, serialize } from '@wordpress/blocks';
import '@testing-library/jest-dom/extend-expect';

/**
 * Internal dependencies.
 */
import { layoutOptions } from '../utilities';
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;

describe( 'coblocks/row', () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );
	} );

	it( 'should render column counts', () => {
		[ 1, 2, 3, 4 ].forEach( ( columnCount ) => {
			block.attributes.columns = columnCount;
			const serializedBlock = serialize( block );
			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( 'data-columns="' + columnCount + '"' );
			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );

	it( 'should render column and layout data', () => {
		// Single column, 100% layout
		block.attributes.columns = 1;
		block.attributes.layout = '100';
		let serializedBlock = serialize( block );
		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'data-columns="1"' );
		expect( serializedBlock ).toContain( 'data-layout="100"' );
		expect( serializedBlock ).toMatchSnapshot();

		// Looping through the layoutOptions to match each column + template
		for ( const key in layoutOptions ) {
			if ( layoutOptions.hasOwnProperty( key ) ) {
				block.attributes.columns = key;

				for ( const ky in layoutOptions[ key ] ) {
					block.attributes.layout = layoutOptions[ key ][ ky ].key;

					serializedBlock = serialize( block );
					expect( serializedBlock ).toBeDefined();
					expect( serializedBlock ).toContain( 'data-columns="' + key + '"' );
					expect( serializedBlock ).toContain( 'data-layout="' + layoutOptions[ key ][ ky ].key + '"' );
					expect( serializedBlock ).toMatchSnapshot();
				}
			}
		}
	} );

	it( 'should apply custom classes with inline css', () => {
		block.attributes.backgroundColor = '#123456';
		block.attributes.backgroundImg = 'http://background.img';
		block.attributes.backgroundType = 'image';
		block.attributes.focalPoint = { x: 0.1, y: 0.25 };
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toContain( 'has-123456-background-color' );
		expect( serializedBlock ).toContain( `url(${ block.attributes.backgroundImg })` );
		expect( serializedBlock ).toContain( 'background-position:10% 25%' );
	} );
} );
