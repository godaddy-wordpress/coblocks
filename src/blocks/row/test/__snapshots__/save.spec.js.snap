// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/row should render column and layout data 1`] = `
"<!-- wp:coblocks/row {\\"columns\\":1,\\"layout\\":\\"100\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"1\\" data-layout=\\"100\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 2`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"2\\",\\"layout\\":\\"50-50\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"2\\" data-layout=\\"50-50\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 3`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"2\\",\\"layout\\":\\"66-33\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"2\\" data-layout=\\"66-33\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 4`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"2\\",\\"layout\\":\\"33-66\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"2\\" data-layout=\\"33-66\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 5`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"2\\",\\"layout\\":\\"75-25\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"2\\" data-layout=\\"75-25\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 6`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"2\\",\\"layout\\":\\"25-75\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"2\\" data-layout=\\"25-75\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 7`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"3\\",\\"layout\\":\\"33-33-33\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"3\\" data-layout=\\"33-33-33\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 8`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"3\\",\\"layout\\":\\"50-25-25\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"3\\" data-layout=\\"50-25-25\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 9`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"3\\",\\"layout\\":\\"25-25-50\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"3\\" data-layout=\\"25-25-50\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 10`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"3\\",\\"layout\\":\\"25-50-25\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"3\\" data-layout=\\"25-50-25\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 11`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"3\\",\\"layout\\":\\"20-60-20\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"3\\" data-layout=\\"20-60-20\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 12`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"4\\",\\"layout\\":\\"25-25-25-25\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"4\\" data-layout=\\"25-25-25-25\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 13`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"4\\",\\"layout\\":\\"40-20-20-20\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"4\\" data-layout=\\"40-20-20-20\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column and layout data 14`] = `
"<!-- wp:coblocks/row {\\"columns\\":\\"4\\",\\"layout\\":\\"20-20-20-40\\"} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"4\\" data-layout=\\"20-20-20-40\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column counts 1`] = `
"<!-- wp:coblocks/row {\\"columns\\":1} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"1\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column counts 2`] = `
"<!-- wp:coblocks/row {\\"columns\\":2} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"2\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column counts 3`] = `
"<!-- wp:coblocks/row {\\"columns\\":3} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"3\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;

exports[`coblocks/row should render column counts 4`] = `
"<!-- wp:coblocks/row {\\"columns\\":4} -->
<div class=\\"wp-block-coblocks-row\\" data-columns=\\"4\\"><div class=\\"wp-block-coblocks-row__inner has-no-padding has-no-margin\\"></div></div>
<!-- /wp:coblocks/row -->"
`;
